### DISCUSSION TITLE
Accessing the HTTP response status code of mutations

### DISCUSSION THREAD

**Discussion by hornta** [reduxjs/redux-toolkit#2709] [Q&A] [answered]
How can I get the HTTP response status code from a mutation response? I need to do some logic depending on the status

**Comment by phryneas** [comment, marked as answer]
You can access `meta.response` in `transformResponse` and make it part of `data` that way.

**Reply by hornta** [reply]
Thanks, I'll do that. I'm certain there is a reason but I still want to ask as to why meta isn't appended on the resolved mutation result like data?

**Reply by phryneas** [reply]
That would mean putting non-serializable data in the store, which would make persistence and SSR impossible.

**Reply by hornta** [reply]
Alright I see. And I assume the meta argument is typed as optional since a custom `baseQueryFn` may not provide any meta? Because I looked at the default `fetchBaseQuery` and there it looks like meta is always an object with a request object but it's typed as `FetchBaseQueryMeta | undefined`.

So could the meta argument in transformResponse be somehow inferred from the baseQueryFn used in createApi or is that not just possible now?

**Reply by phryneas** [reply]
At that point it's a compromise to make the types not explode.

**Reply by be-thomas** [reply]
I'm here on the last day of 2024, Its weird how with all the advancements, still we are not sure how to infer type. Or even looking for small small issues like HTTP Status code damn!, that ate like an hour of my life.

why do devs have to make it harder for every single person? at the very least you need a status code!

Bad decision by open source developers these days. These bad decisions eat up the time off your dev hours to prevent you from actually working on your project.

Sorry to @hornta 
cheering up @phryneas

**Reply by EskiMojo14** [reply]
@be-thomas not every base query will have a concept of a status code, that's an HTTP specific concept. hence, it's included in the base query meta, which is allowed to be a different type per base query.

**Reply by phryneas** [reply]
> Bad decision by open source developers these days. These bad decisions eat up the time off your dev hours to prevent you from actually working on your project.

So you're saying instead of you getting paid for an hour of work on this we should put in a few dozen more hours of work to fix it for you for free? 
You realize that we're not paid for this, right?
PRs are welcome.

Happy new year, I guess.

**Reply by be-thomas** [reply]
> So you're saying instead of you getting paid for an hour of work on this we should put in a few dozen more hours of work to fix it for you for free? 

Hello everyone,

I want to clarify that my previous comment wasn’t meant to offend anyone. I deeply respect the incredible work done by open-source contributors, often unpaid.

My concern is about the rapid changes in web dev libraries—
- For instance, frameworks and libraries like React Router DOM often undergo significant API overhauls within short intervals.
- To make matters trickier, frameworks get deprecated unexpectedly, and developers are left scrambling to migrate. The fact that we now even have meta-frameworks that compile to other frameworks speaks volumes about the need to safeguard against these rapid shifts (Astro compiles to React,Angular,Vue, etc). It’s a testament to the toll this dynamic landscape takes on developers.
`I just hope government restricts the number of javascript frameworks created.`

Small issues, like dealing with HTTP status codes or type inference, can eat into hours of dev time, which is frustrating when trying to focus on actual project work. http status is something used so often that I expected Redux devs to have taken good care of it.

I now understand the design constraints explained here, and I appreciate the effort that goes into these decisions. Thank you for your hard work, and I’ll aim to contribute positively in the future.

Happy New Year! 🎉
Best regards.
Special thanks to @phryneas
